recipes/z3/all/conanfile.py

```diff
@@ -70,39 +83,76 @@ def _configure_cmake(self):
         self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "Visual Studio": "15.7",
+            "clang": "5",
+            "apple-clang": "10",
+        }
+
+    def validate(self):
+        if tools.Version(self.version) >= tools.Version("4.8.11"):
+            if self.settings.compiler.get_safe("cppstd"):
+                tools.check_min_cppstd(self, "17")
+            compiler = self.settings.compiler
+            min_version = self._compilers_minimum_version\
+                .get(str(compiler), False)
+            if min_version:
+                if tools.Version(compiler.version) < min_version:
```

Comment:
```suggestion
                if Version(compiler.version) < min_version:
```
from conan.tools.scm import Version